tests/integration/test_basic.py

```diff
@@ -3,17 +3,244 @@
+        for test in multiple_erroring_tests_for_asc.tests:
+            test_runs = test.qtest_test_runs
+            assert len(test_runs) == 1
+            pytest.helpers.assert_qtest_property(test_runs[0], 'Status', test_run_status_exp)
+
+    # noinspection PyUnresolvedReferences
+    def test_publish_mk8s_multiple_erroring(self, multiple_erroring_tests_for_mk8s):
+        """Verify ZigZag can publish results from the "mk8s" CI environment with multiple erroring tests in the
+        JUnitXML file.
+        """
+
+        # Setup
+        multiple_erroring_tests_for_mk8s.assert_invoke_zigzag()
+
+        # Expectations
+        test_run_status_exp = 'Failed'
+
+        # Test
+        for test in multiple_erroring_tests_for_mk8s.tests:
+            test_runs = test.qtest_test_runs
+            assert len(test_runs) == 1
+            pytest.helpers.assert_qtest_property(test_runs[0], 'Status', test_run_status_exp)
+
+
+class TestSkipping(object):
+    """Test cases for tests with 'Incomplete' status in qTest."""
+
+    # noinspection PyUnresolvedReferences
+    def test_publish_asc_multiple_skipping(self, multiple_skipping_tests_for_asc):
+        """Verify ZigZag can publish results from the "asc" CI environment with multiple skipping tests in the
+        JUnitXML file.
+        """
+
+        # Setup
+        multiple_skipping_tests_for_asc.assert_invoke_zigzag()
+
+        # Expectations
+        test_run_status_exp = 'Incomplete'
+
+        # Test
+        for test in multiple_skipping_tests_for_asc.tests:
+            test_runs = test.qtest_test_runs
+            assert len(test_runs) == 1
+            pytest.helpers.assert_qtest_property(test_runs[0], 'Status', test_run_status_exp)
+
+    # noinspection PyUnresolvedReferences
+    def test_publish_mk8s_multiple_skipping(self, multiple_skipping_tests_for_mk8s):
+        """Verify ZigZag can publish results from the "mk8s" CI environment with multiple skipping tests in the
+        JUnitXML file.
+        """
+
+        # Setup
+        multiple_skipping_tests_for_mk8s.assert_invoke_zigzag()
+
+        # Expectations
+        test_run_status_exp = 'Incomplete'
+
+        # Test
+        for test in multiple_skipping_tests_for_mk8s.tests:
+            test_runs = test.qtest_test_runs
+            assert len(test_runs) == 1
+            pytest.helpers.assert_qtest_property(test_runs[0], 'Status', test_run_status_exp)
+
+
+class TestMixed(object):
+    """Test cases for test suites with 'Passed', 'Failed' and 'Incomplete' test statuses in qTest."""
+
+    # noinspection PyUnresolvedReferences
+    def test_publish_asc_mixed_status(self, mixed_status_tests_for_asc):
+        """Verify ZigZag can publish results from the "asc" CI environment with multiple skipping tests in the
+        JUnitXML file.
+        """
+
+        # Setup
+        mixed_status_tests_for_asc.assert_invoke_zigzag()
+
+        # Expectations
+        test_run_status_exps = ['Passed', 'Failed', 'Failed', 'Incomplete']
```

Comment:
The internal status settings for the tests generated by the factory are managed separately from this file. It would be nice to set the expectation and defined status in the same location to prevent them from changing independently of each other. This may not be possible at the moment given the construction of fixtures, but we might want to flag it as tech-debt.

Reply:
@johnduarte Ahh, yeah that is because the test fixtures and cases assume a certain configuration of the target qTest project which is naive. [I have created a ticket to address the tech debt later.](https://rpc-openstack.atlassian.net/browse/ASC-787)

Reply:
👍 Thanks @ryan-rs .